src/foam/net/Port.js

```diff
@@ -38,10 +40,25 @@ foam.CLASS({
   javaCode: `
   public static int get(X x, String id) {
     String override = System.getProperty(id+".port");
-    if ( ! SafetyUtil.isEmpty(override) ) return Integer.parseInt(override);
+    if ( ! SafetyUtil.isEmpty(override) )
+      return Integer.parseInt(override);
+
     Port port = (Port) ((DAO) x.get("portDAO")).find_(x, id);
     if ( port == null ) {
-      throw new IllegalArgumentException("Port not found");
+      FObject nspec = (FObject) x.get(id);
```

Comment:
This isn't really the nspec, but the actual service, would be more accurate to say

```
FObject service = (FObject)x.get(id)
```